modality-probe-debug-collector/src/cli.rs

```diff
@@ -0,0 +1,666 @@
+        0x00, 0x00, 0x02, 0x00, 0x00, 0x00, 0x04, 0x00, 0x00, 0x00, 0x10, 0x00, 0x00, 0x00, 0x31,
+        0x00, 0x00, 0x00, 0x03, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0xf8, 0x00, 0x00, 0x00, 0x43, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x01, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x3b, 0x00, 0x00, 0x00, 0x03, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x3b, 0x00, 0x00, 0x00, 0x12,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00,
+        0x00,
+    ];
+
+    fn compile_symbol_example() {
+        Command::new("cargo")
+            .arg("build")
+            .current_dir(canonicalize("./tests/symbols-example").unwrap())
+            .output()
+            .unwrap();
+    }
+
+    fn create_be_example() -> TempPath {
+        let mut file = NamedTempFile::new().unwrap();
+        file.as_file_mut().write(&BE_BIN_RAW).unwrap();
+        file.into_temp_path()
+    }
+
+    fn options_from_str(input: &str) -> CLIOptions {
+        CLIOptions::from_iter_safe(input.split(" ")).unwrap()
+    }
+
+    /// Test basic parsing
+    #[test]
+    fn test_basic() {
+        assert_eq!(
+            config_from_options(options_from_str(
+                "modality-probe-debug-collector \
+                --session-id 0 \
+                --little-endian \
+                --attach stm32 \
+                --interval 1s \
+                --output-file ./out \
+                0x100"
+            ))
+            .unwrap(),
+            Config {
+                session_id: 0.into(),
+                big_endian: false,
+                attach_target: Some("stm32".to_string()),
+                gdb_addr: None,
+                interval: Duration::from_millis(1000),
+                output_path: "./out".into(),
+                probe_addrs: vec![ProbeAddr::Addr(0x100)]
+            }
+        )
+    }
+
+    /// Specify gdb server instead of attach target
+    #[test]
+    fn specify_gdb_server() {
+        assert_eq!(
+            config_from_options(options_from_str(
+                "modality-probe-debug-collector \
+                --session-id 0 \
+                --little-endian \
+                --gdb-addr 127.0.0.1:3000 \
+                --interval 1s \
+                --output-file ./out \
+                0x100"
+            ))
+            .unwrap(),
+            Config {
+                session_id: 0.into(),
+                big_endian: false,
+                attach_target: None,
+                gdb_addr: Some(SocketAddrV4::from_str("127.0.0.1:3000").unwrap()),
+                interval: Duration::from_millis(1000),
+                output_path: "./out".into(),
+                probe_addrs: vec![ProbeAddr::Addr(0x100)]
+            }
+        )
+    }
+
+    /// Should error if both attach target and gdb server specified
+    #[test]
+    #[should_panic]
+    fn error_on_multiple_targets() {
```

Comment:
I generally discourage the use of `should_panic` because it makes it more likely for the wrong sort of error tossed early to masquerade as a desired panic later in a function. 

Clearly that's less pressing of a concern for one-liners like this, but as a general practice, I would recommend doing a pattern of:

```rust
assert!(operation_returning_a_result().is_err());
```
which works simply enough when you don't have precise error management,
but can be upgraded in a straightforward manner when specific error variants
are exposed, e.g.

```rust
assert_eq!(Err(MyFancyErr:DesiredVariant)), operation_returning_a_result());
```